src/modules/utils.js

```diff
@@ -28,5 +28,21 @@ module.exports = {
    */
   timeoutPromise: (timer) => new Promise((resolve) => {
     setTimeout(resolve, timer);
-  })
+  }),
+
+  /**
+   * A function that allows to emit warnings for a specified code. The idea is to
+   * limit a warning emission to one per a specific code.
+   * @param warning - {code: string, detail: string, message: string}
+   */
+  emitWarn: function emitWarn(warning) {
```

Comment:
we also printing deprecation message here: https://github.com/bringg/node-arnavmq/blob/23be17bf225d454b0dc91b63da10baa20a52a9d2/src/index.js#L49 I think it worth using this func there as well.